src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java

```diff
@@ -0,0 +1,207 @@
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Bucket update(Bucket bucket) {
+        try (PreparedStatement statement = connection.prepareStatement(UPDATE_BUCKET_QUERY)) {
+            statement.setLong(2, bucket.getUserId());
+            statement.setLong(3, bucket.getBucketId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("Cannot update bucket with id " + bucket.getBucketId() + e);
+        }
+        return bucket;
+    }
+
+    @Override
+    public boolean deleteById(Long bucketId) {
+        try (PreparedStatement statement = connection.prepareStatement(DELETE_BUCKET_BY_ID)) {
+            statement.setLong(1, bucketId);
+            statement.executeUpdate();
+            return true;
+        } catch (SQLException e) {
+            logger.error("Cannot delete bucket with id " + bucketId + e);
+        }
+        return false;
+    }
+
+    @Override
+    public boolean delete(Bucket bucket) {
+        return deleteById(bucket.getBucketId());
+    }
+
+    @Override
+    public Optional<Bucket> getByUserId(Long userId) {
+        try (PreparedStatement statement = connection.prepareStatement(GET_BUCKET_BY_USER_ID)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return get(resultSet.getLong("bucket_id"));
+            }
+        } catch (SQLException e) {
+            logger.error("Can't get bucket by user's id " + userId + e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void addItem(Bucket bucket, Item item) {
+        try (PreparedStatement statement = connection.prepareStatement(ADD_ITEM_TO_BUCKET_QUERY)) {
+            statement.setLong(1, bucket.getBucketId());
+            statement.setLong(2, item.getItemId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("Cannot add item to bucket " + e);
+        }
+    }
+
+    @Override
+    public void deleteItem(Bucket bucket, Item item) {
+        try (PreparedStatement statement =
+                     connection.prepareStatement(DELETE_ITEM_FROM_BUCKET_QUERY)) {
+            statement.setLong(1, bucket.getBucketId());
+            statement.setLong(2, item.getItemId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("Cannot delete item from bucket with id ="
+                    + bucket.getBucketId() + e);
+        }
+    }
+
+    @Override
+    public void clear(Bucket bucket) {
+        try (PreparedStatement statement = connection.prepareStatement(CLEAR_BUCKET_QUERY)) {
+            statement.setLong(1, bucket.getBucketId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("Cannot clear bucket with id = " + bucket.getBucketId());
```

Comment:
```suggestion
            logger.error("Cannot clear bucket with id = " + bucket.getBucketId(), e);
```
скрізь потрібно передавати помилку

Reply:
my fault, done